node/pkg/watchers/hedera/watcher.go

```diff
@@ -0,0 +1,612 @@
+package hedera
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/watchers/evm/connectors/ethabi"
+	abi2 "github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/certusone/wormhole/node/pkg/p2p"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+
+	eth_common "github.com/ethereum/go-ethereum/common"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/readiness"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	"github.com/tidwall/gjson"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+// HCS Mirror Node gRPC Endpoints (urlWS):
+// PREVIEWNET: hcs.previewnet.mirrornode.hedera.com:5600
+// TESTNET: hcs.testnet.mirrornode.hedera.com:5600
+// MAINNET: mainnet-public.mirrornode.hedera.com:443
+
+// urlLCD endpoints
+// MAINNET BASEURL
+// https://mainnet-public.mirrornode.hedera.com/
+// TESTNET BASEURL
+// https://testnet.mirrornode.hedera.com/
+// PREVIEWNET BASEURL
+// https://previewnet.mirrornode.hedera.com/
+
+// {
+//   "anonymous": false,
+//   "inputs": [
+//     {
+//       "indexed": true,
+//       "internalType": "address",
+//       "name": "sender",
+//       "type": "address"
+//     },
+//     {
+//       "indexed": false,
+//       "internalType": "uint64",
+//       "name": "sequence",
+//       "type": "uint64"
+//     },
+//     {
+//       "indexed": false,
+//       "internalType": "uint32",
+//       "name": "nonce",
+//       "type": "uint32"
+//     },
+//     {
+//       "indexed": false,
+//       "internalType": "bytes",
+//       "name": "payload",
+//       "type": "bytes"
+//     },
+//     {
+//       "indexed": false,
+//       "internalType": "uint8",
+//       "name": "consistencyLevel",
+//       "type": "uint8"
+//     }
+//   ],
+//   "name": "LogMessagePublished",
+//   "type": "event"
+// },
+
+type (
+	// Watcher is responsible for looking over a cosmwasm blockchain and reporting new transactions to the contract
```

Comment:
Nit: "hedera", not "cosmwasm". Maybe search for other references. Also uses of "LCD". I don't think that's what it is in this case?

Reply:
Fixed